### PR TITLE
Save query dialog validation

### PR DESCRIPTION
**Pull request by myabc**

### CODE REVIEW

**Comment by myabc** [comment]
@manwithtwowatches I know you also began working on this same issue last week (in 6a17a391). I handled validations slightly differently (namely, not touching the controller). Let me know if you're agreement with this approach.

File: public/templates/work_packages/modals/save.html [hunk 1]
```diff
@@ -4,14 +4,21 @@
...
-    </div>
-    <div>
-      <button class="button_highlight" ng-click="saveQueryAs(queryName)">Save</button>
-      <button class="button" ng-click="modal.closeMe()">Cancel</button>
-    </div>
+    <form name="modalSaveForm">
+      <div>
+        <label>
+          Name:
+          <span class="required"> *</span>
```
Comment (myabc) [inline public/templates/work_packages/modals/save.html:11]: To apply this blue asterisk, I'd love to be able to use a `::before` psuedo-selector here. Unfortunately, this doesn't work with `<input>` elements. Another solution might be to build our own directive for form controllers, but that's too much work for the moment.

File: Gemfile [hunk 2]
```diff
@@ -112,7 +112,7 @@ gem 'sass-rails',        git: 'https://github.com/guilleiguaran/sass-rails.git',
 gem 'sass',             '~> 3.3.6'
 gem 'bourbon',          '~> 4.0'
 gem 'uglifier',         '>= 1.0.3', require: false
-gem 'compass',          '1.0.0.alpha.19'
+gem 'compass',          '1.0.0.alpha.19', require: 'compass/core'
```
Comment (manwithtwowatches) [inline Gemfile:115]: @myabc why is this now necessary?

Comment (myabc) [inline Gemfile:115]: @manwithtwowatches see my commit message in 263ee137. The prerelease versions of `compass-rails` doesn't seem to behaving as it should…

**Pull request merged**